source/ed/data.cpp

```diff
@@ -735,28 +735,54 @@ EdTZWrapper::get_dst_periods(const boost::gregorian::date_period& validity_perio
     }
 
     BOOST_ASSERT(! years.empty());
+    bool is_dst_start_smaller = true;
 
     std::vector<PeriodWithUtcShift> res;
     for (int year: years) {
+        is_dst_start_smaller = (boost_timezone->dst_local_start_time(year).date() < boost_timezone->dst_local_end_time(year).date());
```

Comment:
It would be nice to have `boost_timezone->dst_local_start_time(year).date()` and `boost_timezone->dst_local_end_time(year).date()` in variables like `saving_time_start` and `normal_time_start` (stored depending on `is_dst_start_smaller`).
This would avoid almost all of the if/else added